community/detectors/selenium_grid_rce_via_exposed_server/src/main/java/com/google/tsunami/plugins/detectors/rce/selenium/RCEViaExposedSeleniumGridDetector.java

```diff
@@ -0,0 +1,523 @@
+      commandToInject = String.format("curl -o %s file:///dev/null", RCE_TEST_FILE_PATH);
+      executeCommandViaChrome(networkService, commandToInject);
+      return true;
+
+    } else {
+      // Not vulnerable
+      logger.atInfo().log(
+          "File %s doesn't exist, or doesn't contain %s string",
+          RCE_TEST_FILE_PATH, RCE_TEST_STRING);
+      return false;
+    }
+  }
+
+  private static String buildTargetUrl(NetworkService networkService, String path) {
+    StringBuilder targetUrlBuilder = new StringBuilder();
+
+    if (NetworkServiceUtils.isWebService(networkService)) {
+      targetUrlBuilder.append(NetworkServiceUtils.buildWebApplicationRootUrl(networkService));
+
+    } else {
+      // Default to HTTP protocol when the scanner cannot identify the actual service.
+      // HTTP is also used in a default Selenium Grid install.
+      targetUrlBuilder
+          .append("http://")
+          .append(toUriAuthority(networkService.getNetworkEndpoint()))
+          .append("/");
+    }
+    targetUrlBuilder.append(path);
+    return targetUrlBuilder.toString();
+  }
+
+  // Verifies that Selenium Grid is exposed.
+  // Password-protected Selenium Grid will issue a 401 Unauthorized response with header: 
+  // WWW-Authenticate: Basic realm="selenium-server"
+  private boolean isSeleniumGridExposed(NetworkService networkService) {
+    String statusUri = buildTargetUrl(networkService, SELENIUM_GRID_SERVICE_PATH + "/status");
+
+    try {
+      HttpResponse response =
+          httpClient.send(get(statusUri).withEmptyHeaders().build(), networkService);
+      return (response.status().isSuccess()
+          && response.bodyString().map(body -> body.contains("Selenium Grid")).orElse(false));
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log("Request to target %s failed", statusUri);
+    }
+
+    return false;
+  }
+
+  // Ensures Selenium is in ready state and accepts new requests to avoid stuck requests.
+  // Returns true when ready, or false on timeout or failure
+  private boolean isSeleniumGridReady(NetworkService networkService) {
+    boolean seleniumIsReady = false;
+    String statusUri = buildTargetUrl(networkService, SELENIUM_GRID_SERVICE_PATH + "/status");
+
+    logger.atInfo().log(
+        "Waiting for Selenium Grid to enter ready state (timeout is %d s)",
+        (POLLING_RATE * POLLING_ATTEMPTS / 1000));
+    int attempts = 0;
+
+    // Request Selenium Grid ready status until true, or the number of attempts get exhausted
+    while (true) {
+      attempts++;
+      if (attempts > POLLING_ATTEMPTS) {
+        logger.atWarning().log("Timeout while waiting for Selenium to become ready");
+        return false;
+      }
+
+      try {
+        HttpResponse response =
+            httpClient.send(get(statusUri).withEmptyHeaders().build(), networkService);
+
+        if (response.status().isSuccess() || response.bodyJson().isPresent()) {
```

Comment:
Is this `or` condition the intended one? As I understand this code, the server response may be != 200 but still have a json body. 
Also, **every** other case, including an occurred exception, would return a false value stopping the polling (just asking if I understood correctly)